packages/web3/src/utils/bs58.test.ts

```diff
@@ -16,12 +16,59 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { isBase58 } from './bs58'
+import { isBase58, base58ToBytes, bs58 } from './bs58'
+import { TraceableError } from '../error'
 
 describe('bs58', () => {
-  it('should validate bs58 string', () => {
-    expect(isBase58('32UWxgjUHwH7P1J61tb12')).toEqual(true)
-    expect(isBase58('32U.WxgjUHwH7P1J61tb12')).toEqual(false)
-    expect(isBase58('')).toEqual(false)
+  describe('isBase58', () => {
+    it('should validate valid bs58 strings', () => {
+      expect(isBase58('32UWxgjUHwH7P1J61tb12')).toBe(true)
+      expect(isBase58('1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2')).toBe(true)
+    })
+
+    it('should invalidate strings with non-base58 characters', () => {
+      expect(isBase58('32U.WxgjUHwH7P1J61tb12')).toBe(false)
+      expect(isBase58('Invalid_Base58')).toBe(false)
+    })
+
+    it('should handle empty strings', () => {
+      expect(isBase58('')).toBe(false)
+    })
+
+    it('should handle whitespace strings', () => {
+      expect(isBase58('   ')).toBe(false)
+      expect(isBase58('\n\t')).toBe(false)
+    })
+  })
+
+  describe('base58ToBytes', () => {
+    it('should convert valid base58 strings to bytes', () => {
+      const result = base58ToBytes('32UWxgjUHwH7P1J61tb12')
+      expect(result).toBeInstanceOf(Uint8Array)
+      expect(result.length).toBeGreaterThan(0)
```

Comment:
Perhaps we can check `bs58.encode(result)` would give us `32UWxgjUHwH7P1J61tb12` instead.

Reply:
> Perhaps we can check `bs58.encode(result)` would give us `32UWxgjUHwH7P1J61tb12` instead.

I have added a test to ensure that encoding the result gives us back the original input.